src/Symfony/Component/Mailer/Bridge/Sendgrid/Transport/SendgridApiTransport.php

```diff
@@ -145,6 +145,10 @@ private function getPayload(Email $email, Envelope $envelope): array
             $personalization['custom_args'] = $customArguments;
         }
 
+        foreach ($envelope->getOptions() as $name => $value) {
+            $payload[$name] = $value;
```

Comment:
This allows overriding any pre-existing payload key. This might be too risky.

Reply:
Should we do this instead?

```suggestion
            if (!array_key_exists($name, $payload) {
                $payload[$name] = $value;
            }
```